### ISSUE TITLE
Output blank for web pages

### ISSUE THREAD

**Issue by Leseratte10** [bazukas/obs-qtwebkit#20]
I have compiled QtWebKit Browser with (and for) Linux, but when I add a new QtWebKit source in OBS, it just shows a transparent rectangle with nothing in it - no matter what web page I use. 

When I use a local HTML file as source, it is rendered perfectly - but when I enter an actual web page, I do not even see a request being made in Wireshark ...

**Comment by Leseratte10** [comment]
The plugin seems to be using the wrong network interface (loopback?): When I enter the IP of my computer (which has a web server running and working), I see nothing, just as for the URL. 

But when I enter 127.0.0.1 (localhost) I see the output of my web server ...

**Comment by bazukas** [comment]
Very strange. I don't handle network communication directly, it is done through qt. Are you sure you don't see any tcp connections being made? You might try forcing renderer process to use specific network interface, I think you can do that on linux.

**Comment by Leseratte10** [comment]
I don't see any connection at all. 
I've started Wireshark, started recording on interface "any" (so, everything). 

Then I set a filter "ip.addr==127.0.0.1", changed the URL in OBS to "http://127.0.0.1/test" and I see a HTTP request and my web page in OBS. 

However, with a filter of "ip.addr==216.58.210.14" (Google) and an URL "http://216.58.210.14/test", nothing happens. The screen in OBS stays blank, and I don't see a single outgoing packet to that address.

**Comment by bazukas** [comment]
Unfortunately, I'm unable to replicate the issue on my machine.
I assume that this does not happen in normal browsers? What about QubZilla, I think it is based on QtWebkit too.
Is there anything weird in your network interfaces, or everything's normal?

**Comment by Leseratte10** [comment]
This does not happen in "normal" Browsers (Firefox & Chromium), but it does indeed happen in QupZilla - exactly the same, localhost works, other IPs do not. 

Well, regarding the question if there is anything weird in my interfaces, that depends on how you define "weird". I do have a few static routes - 

`$ route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.0.1.1        0.0.0.0         UG    0      0        0 eth0
10.0.1.0        0.0.0.0         255.255.255.0   U     0      0        0 eth0
10.0.20.0       0.0.0.0         255.255.255.0   U     0      0        0 eth0
169.254.0.0     0.0.0.0         255.255.0.0     U     1000   0        0 eth0`

but there is a normal 0.0.0.0-route over my LAN connection to my gateway 10.0.1.1. 

I also have a secondary IP address on my interface eth0: 

`$ ifconfig
eth0      Link encap:Ethernet  HWaddr 94:xx:xx:xx:xx:f4  
          inet addr:10.0.1.104  Bcast:10.0.1.255  Mask:255.255.255.0
          inet6 addr: 2001:xxxx:xxxx:0:xxxx:xxff:fexx:xxxx/64 Scope:Global
          inet6 addr: 2001:xxxx:xxx::1/64 Scope:Global
          inet6 addr: fe80::xxxx:xxff:fexx:xxxx/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:23563309 errors:2 dropped:19 overruns:0 frame:1
          TX packets:16473370 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:26071624222 (26.0 GB)  TX bytes:9074296552 (9.0 GB)
          Interrupt:20 Memory:f3200000-f3220000 

eth0:1    Link encap:Ethernet  HWaddr 94:xx:xx:xx:xx:f4  
          inet addr:10.0.20.1  Bcast:10.0.20.255  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          Interrupt:20 Memory:f3200000-f3220000 `

Maybe one of those two things causes problems?

**Comment by bazukas** [comment]
Unfortunately, I don't think I'll be able to help you with this problem, but it seems like that the problem is on Qts side. So you might find some help with this problem on QupZilla or Qt support forums. Sorry I couldn't help resolving your issue. 

Let me know if you find anything.

**Comment by DragOnGamer626** [comment]
What port are you using, what OS and who's your ISP? I'm asking because I have AT&T and when I used to run any Linux OS, I had to disable IPV6 and also connect to all local servers at http://localhost:<insert your port here>

Hell, on Windows, for OBS-Studio, I have to connect to localhost anyways to get my chat overlay to work properly.

**Comment by Leseratte10** [comment]
Port is obviously 80 (normal web page), ISP is Unitymedia / Liberty Global. 

I also have IPv6 enabled on my machine, but I cannot just disable that since it is needed for other services / applications. I can do a test if disabling IPv6 solves this issue, but disabling IPv6 can not be the final solution.

**Comment by DragOnGamer626** [comment]
I've never heard of that ISP but AT&T at the time blocked all IPv6 connections on my Linux boxes and there was really no way around it. This was about 6 years ago, though, so it maybe different now.

Have you tried a port like 8080 which is usually not used by anything?

**Comment by Leseratte10** [comment]
But I'd like to use my normal Apache webserver on Port 80 ...
A working webserver runs on both IPv4 and IPv6 on Port 80 and can be reached from the local network or from the public internet both over IPv4 and IPv6, so my provider isn't blocking anything.

**Comment by DragOnGamer626** [comment]
Don't know what to tell you then. Was just offering my own experiences. You could still try a different port not because of your ISP but because something else might be using it

**Comment by Leseratte10** [comment]
Sure something is using the port, otherwise I could not load the web page.

**Comment by DragOnGamer626** [comment]
Of course, have you tried this fork? https://github.com/vityafx/obs-qtwebkit

It has a patch that's not in the main repo that might solve your problem.

**Comment by DragOnGamer626** [comment]
Some people were having issues getting notifications and stuff to work and that's what the patch was supposed to be for. I can't test it myself since I don't have a Linux box setup atm

**Comment by Leseratte10** [comment]
That patch only fixes something with the UA since twitchAlerts had problems with non-standard User Agents. But there is not even an outgoing connection to my server, so this patch will not help.

**Comment by DragOnGamer626** [comment]
Fair enough. Just throwing ideas out there.

**Comment by DragOnGamer626** [comment]
What OS are you on, btw?

**Comment by Leseratte10** [comment]
Ubuntu 15.10.

**Comment by DragOnGamer626** [comment]
Have you thought about updating to 16.04? It's an LTS version which means it's considered the latest stable and recommend version

**Comment by Leseratte10** [comment]
I bet that doesn't fix the problem...

**Comment by DragOnGamer626** [comment]
It might, though. Ubuntu doesn't like to keep older versions up to date with software patches as much as they really should

**Comment by DragOnGamer626** [comment]
That's one of the main reasons I eventually switched to Gentoo and Arch. Both have better update management and really let you get bug fixes at the price of having to learn how to really work with the internals of your system

**Comment by DragOnGamer626** [comment]
One last thing; this might not work, but here's a fork of the original OBS-Browser plugin that is supposed to be compilable on Linux: https://github.com/OsirisNL/obs-browser

Can you try that repo?

**Comment by Leseratte10** [comment]
Compiling this fails with gtk/gtk.h not found although I have installed libgtk-3-dev and libgtk2.0-dev.

**Comment by DragOnGamer626** [comment]
That error means that you need to set it to use GTK-2 instead of GTK-3 (which is the Ubuntu default). There's a couple of ways you can do that, but that's beyond what I can tell you atm. Also, there's a Linux-support branch which you'll probably have to change to get it to compile.

Now, if you hold on, I can quickly setup Kubuntu in Virtualbox and see if I could be of more help with this issue.

**Comment by DragOnGamer626** [comment]
Also, mind you, my Kubuntu version will be 16.04 not 15.10. That might affect the outcome, especially since I'm using it on a virtual machine. I also won't have a webserver setup for it, per se. At least not like your setup. Instead, I'll be testing it out with the http://www.restream.io chat web server. I multistream and that program has the ability to run chat through a local web server @ port 8000-9000 (depending on your settings).

**Comment by TheAifam5** [comment]
That can help: https://github.com/vityafx/obs-qtwebkit/pull/1

**Comment by bazukas** [comment]
Please see https://github.com/bazukas/obs-linuxbrowser
